cinema/views.py

```diff
@@ -101,8 +103,26 @@ def get_serializer_class(self):
         if self.action == "retrieve":
             return MovieDetailSerializer
 
+        if self.action == "upload_image":
+            return MovieImageSerializer
+
         return MovieSerializer
 
+    @action(
+        methods=["POST"],
+        detail=True,
+        url_path="upload-image",
+        permission_classes=[IsAdminUser, ]
```

Comment:
I think you don't need permission_classes here

Reply:
i think its real important, because no-admin user can't add image to movie

Reply:
you already have such permission in this view

Reply:
I also think that you don't need permission_classes here